TheMET/TheMET/Network/MetAPI.swift

```diff
@@ -19,22 +19,22 @@ class MetAPI{
         self.networkManager = networkManager
     }
     
-    func objects(metadataDate: Date? = nil, departmentIds: [Int] = [], completion: @escaping (ObjectsResponse?) -> Void) {
+    func objects(metadataDate: Date? = nil, departmentIds: [Int] = [], completion: @escaping (Result<ObjectsResponse, MetAPIError>) -> Void) {
         self.metAPICache.objects(metadataDate: metadataDate, departmentIds: departmentIds) { [weak self] objectsResponce in
             guard let self = self else {
-                completion(nil)
+                completion(.failure(.metAPIdoesNotAnswer))
                 return
             }
             if let objectsResponce = objectsResponce {
-                completion(objectsResponce)
+                completion(.success(objectsResponce))
             } else {
                 self.executeObjects(metadataDate: metadataDate, departmentIds: departmentIds) {[weak self] realResponse in
                     guard let realResponse = realResponse else {
-                        completion(nil)
+                        completion(.failure(.noDataInResponce))
```

Comment:
You still rely on `nil`/`non-nil` state of the `realResponse`. Please introduce `Result` to `func executeObjects` as well